pkg/segment/structs/evaluationstructs.go

```diff
@@ -1784,20 +1815,81 @@ func (self *TextExpr) EvaluateText(fieldToValue map[string]utils.CValueEnclosure
 			return "", fmt.Errorf("TextExpr.EvaluateText: failed to compile regex '%s': %v", regexStr, err)
 		}
 		return regex.ReplaceAllString(baseStr, replacementStr), nil
-	case "mvappend":
-		fallthrough
+	case "mvindex":
+		mvSlice, err := self.validateAndExtractMultiValueFields(fieldToValue)
+		if err != nil {
+			return "", fmt.Errorf("TextExpr.EvaluateText: %v", err)
+		}
+		if self.StartIndex == nil {
+			return "", fmt.Errorf("TextExpr.EvaluateText: self.StartIndex is required but is nil")
+		}
+		startIndex, err := strconv.Atoi(self.StartIndex.Value)
+		if err != nil {
+			return "", fmt.Errorf("TextExpr.EvaluateText: failed to parse startIndex: %v", err)
+		}
+
+		if startIndex < 0 {
+			startIndex += len(mvSlice)
+		}
+		// If endIndex is not provided, use startIndex as endIndex to fetch single value
+		endIndex := startIndex
+		if self.EndIndex != nil {
+			endIndex, err = strconv.Atoi(self.EndIndex.Value)
+			if err != nil {
+				return "", fmt.Errorf("TextExpr.EvaluateText: failed to parse endIndex: %v", err)
+			}
+			if endIndex < 0 {
+				endIndex += len(mvSlice)
+			}
+		}
+		// Check for index out of bounds
+		if startIndex > endIndex || startIndex >= len(mvSlice) || endIndex < 0 || endIndex >= len(mvSlice) {
+			return "NULL", nil
+		}
+
+		return strings.Join(mvSlice[startIndex:endIndex+1], ","), nil
+	case "mvjoin":
+		mvSlice, err := self.validateAndExtractMultiValueFields(fieldToValue)
+		if err != nil {
+			return "", fmt.Errorf("TextExpr.EvaluateText: %v", err)
+		}
+
+		if self.Delimiter == nil {
+			return "", fmt.Errorf("TextExpr.EvaluateText: Delimiter is nil")
+		}
+		delimiter := self.Delimiter.RawString
+
+		return strings.Join(mvSlice, delimiter), nil
 	case "mvcount":
+		mvSlice, err := self.validateAndExtractMultiValueFields(fieldToValue)
+		if err != nil {
+			return "", fmt.Errorf("TextExpr.EvaluateText: %v", err)
+		}
+
+		count := len(mvSlice)
+
+		return strconv.Itoa(count), nil
+	case "mvfind":
+		mvSlice, err := self.validateAndExtractMultiValueFields(fieldToValue)
+		if err != nil {
+			return "", fmt.Errorf("TextExpr.EvaluateText: %v", err)
+		}
+		regex := self.Regex
+
+		for index, value := range mvSlice {
+			if regex.GetCompiledRegex().MatchString(value) {
+				return strconv.Itoa(index), nil
+			}
+		}
+
+		// If no match is found
+		return "NULL", nil
```

Comment:
Does Splunk return the string `"NULL"` here, or some other representation of NULL? Will our later processing correctly handle the string `"NULL"` (e.g., if we pipe this into the `isnull()` SPL command)?

Same comment for line 1847

Reply:
I just checked on Splunk. Splunk doesn't return the string `"NULL"`; instead, it returns an empty value in the field. The `isNull` function works fine on Splunk, but it doesn't work as expected in our case because our implementation of `isNull` checks for `DType` `Backfill` and not a string representation of `null`. Since this implementation is in EvaluateText, we cannot return `null` or set the `DType` to `Backfill`, as it will be overridden before it goes to the processing of `isNull` and it will be processed as the string that is returned by the `EvaluateText` method. How do you think we should handle this? Should we return an empty string and update `isNull` to consider empty strings as `null` values? or return the string `"NULL"` and update `isNull` to check for `backfill` or string `"NULL"` to determine if the value is null.

Reply:
I think there's two ok options:
1. As you suggested, return an empty string and update our code to treat that the same as null
2. Create a new generic type like `Option[T]` that will either store a value or indicate that there's no value there. (Similar to `Option` in Rust, if you're familiar). Then to return a NULL string we'd just return a `Option[string]` with no string value.

Option 1 (and similar) will be faster to implement, but leaves you with an small bug that you can never really remove until you do Option 2 (or similar).

I think that unless you really want to do Option 2, you can do Option 1 for now

Reply:
Sounds good, Andrew. I looked into the implementation of Option 2, but I believe it requires updating the return type and the flow currently implemented. We can adopt this flow and use it for all other commands that can return `null` so that `isnull` can recognize them as `null`. This approach is not limited to `mvindex`; it applies to other commands as well where we return `null`. We'll need to update those to follow the same approach as Option 2. 

For now, I have pushed a commit for Option 1.